libraries/botbuilder-adapter-twitter/src/twitter.ts

```diff
@@ -0,0 +1,195 @@
+import { Activity, ActivityTypes, BotAdapter, TurnContext, ConversationReference, ResourceResponse, WebRequest, WebResponse } from 'botbuilder';
+import * as Twitter from 'twitter';
+import { TwitterSettings } from './schema';
+import { retrieveBody } from './util';
+
+/**
+ * @module botbuildercommunity/adapter-twitter
+ */
+
+function createTwitterClient(settings: TwitterSettings): Twitter {
+    return new Twitter(settings);
+}
+
+export class TwitterAdapter extends BotAdapter {
+
+    protected readonly settings: TwitterSettings;
+    protected readonly client: any;
+    protected readonly channel: string = 'twitter';
+
+    public constructor(settings: TwitterSettings) {
+        super();
+        this.settings = settings;
+        try {
+            this.client = this.createTwitterClient(settings);
+        }
+        catch (e) {
+            throw new Error(`Error creating Twitter client: ${ e.message }.`);
+        }
+    }
+
+    public async sendActivities(context: TurnContext, activities: Partial<Activity>[]): Promise<ResourceResponse[]> {
+        const responses: ResourceResponse[] = [];
+        for (let i = 0; i < activities.length; i++) {
+            const activity: Partial<Activity> = activities[i];
+
+            switch (activity.type) {
+                case ActivityTypes.Message:
+                    if (!activity.conversation || !activity.conversation.id) {
+                        throw new Error(`Activity doesn't contain a conversation id.`);
+                    }
+                    try {
+                        const message = this.parseActivity(activity);
+                        const res: any = await new Promise((resolve, reject): any => {
```

Comment:
await new Promise should not be necessary. You should be able to do 
`await this.client.post`.

Reply:
Although the Twitter library I'm using allows for Promise-based usage if no callback is present, it doesn't give you access to the raw response--only the resulting tweet. If you want the raw response, you need to use the callback-based version, which I've wrapped in a Promise here.